### PR TITLE
Clarify genesis handling logic

### PR DESCRIPTION
**Pull request by tudor-malene**
Removed some code that was leftover from debugging. 
Minor refactored and commented some key logic.

### CODE REVIEW

**Review by joeldudley — APPROVED** [review]
LGTM

File: go/obscuronode/enclave/state.go [hunk 1]
```diff
@@ -118,6 +118,33 @@ func updateState(b *types.Block, s Storage, blockResolver BlockResolver) *blockS
...
+	bs := calculateBlockState(b, parentState, s, blockResolver, rollups)
+
+	s.SetBlockState(b.Hash(), bs)
+
+	return bs
+}
+
+func handleGenesisRollup(b *types.Block, s Storage, rollups []*Rollup, genesisRollup *Rollup) (genesisState *blockState, isGenesis bool) {
 	// the incoming block holds the genesis rollup
 	// todo change this to an hardcoded hash on testnet/mainnet
```
Comment (otherview) [inline go/obscuronode/enclave/state.go:149]:
```suggestion
	// the incoming block holds the genesis rollup
	// calculate and return the new block state
	// todo change this to an hardcoded hash on testnet/mainnet
```

File: go/obscuronode/enclave/state.go [hunk 2]
```diff
@@ -134,30 +161,14 @@ func updateState(b *types.Block, s Storage, blockResolver BlockResolver) *blockS
 			foundNewRollup: true,
 		}
 		s.SetBlockState(b.Hash(), &bs)
-		return &bs
+		return &bs, true
 	}
 
 	// re-processing the block that contains the rollup
```
Comment (otherview) [inline go/obscuronode/enclave/state.go:167]: Can we add more detail to this comment ?

**Review by otherview — COMMENTED** [review]
Looks good! I'd add a few more comments to make some ops clearer.

File: go/obscuronode/enclave/state.go (hunk 1 above)
Comment (tudor-malene) [reply go/obscuronode/enclave/state.go:149]: didn't know you could suggest changes. this is cool

File: go/obscuronode/enclave/state.go (hunk 2 above)
Comment (tudor-malene) [reply go/obscuronode/enclave/state.go:168]: what would you have in mind?

**Pull request merged**